src/common/CommandLine.cs

```diff
@@ -0,0 +1,1820 @@
+                    else
+                        throw new CommandLineParserException("No command given.");
+                }
+            }
+            return true;
+        }
+
+        /// <summary>
+        /// Return the string representing the help for a single paramter set.  If displayGlobalQualifiers is
+        /// true than qualifiers that apply to all parameter sets is also included, otherwise it is just the
+        /// parameters and qualifiers that are specific to that parameters set. 
+        /// 
+        /// If the parameterSetName null, then the help for the entire program (all parameter
+        /// sets) is returned.  If parameterSetName is not null (empty string is default parameter set),
+        /// then it generates help for the parameter set specified on the command line.
+        /// 
+        /// Since most of the time you don't need help, helpInformatin is NOT collected during the Define* calls
+        /// unless HelpRequested is true.   If /? is seen on the command line first, then this works.  You can
+        /// also force this by setting HelpRequested to True before calling all the Define* APIs. 
+        /// 
+        /// </summary>
+        public string GetHelp(int maxLineWidth, string parameterSetName = null, bool displayGlobalQualifiers = true)
+        {
+            Debug.Assert(_mustParseHelpStrings);
+            if (!_mustParseHelpStrings)          // Backup for retail code.  
+                return null;
+
+            if (parameterSetName == null)
+                return GetFullHelp(maxLineWidth);
+
+            // Find the beginning of the parameter set parameters, as well as the end of the global parameters
+            // (Which come before any parameters set). 
+            int parameterSetBody = 0;         // Points at body of the parameter set (parameters after the parameter set)
+            CommandLineParameter parameterSetParameter = null;
+            for (int i = 0; i < _parameterDescriptions.Count; i++)
+            {
+                CommandLineParameter parameter = _parameterDescriptions[i];
+                if (parameter.IsParameterSet)
+                {
+                    parameterSetParameter = parameter;
+                    if (string.Compare(parameterSetParameter.Name, parameterSetName, StringComparison.OrdinalIgnoreCase) == 0)
+                    {
+                        parameterSetBody = i + 1;
+                        break;
+                    }
+                }
+            }
+
+            if (parameterSetBody == 0 && parameterSetName != String.Empty)
+                return String.Empty;
+
+            // At this point parameterSetBody and globalParametersEnd are properly set. Start generating strings
+            StringBuilder sb = new StringBuilder();
+
+            // Create the 'Usage' line;
+            // TODO: Find a better way to obtain this information.
+#if COREFX
+            string appName = "<AppName>";
```

Comment:
It is probably worth adding a overload to this method that passes in the AppName.